spec/ssh_scan/rule_engine_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rspec'
+require 'ssh_scan/rule_engine'
+
+describe SSHScan::RuleEngine do
+  context "for conditionals" do
+    it "eq, neq should work for any literal" do
+      result = {:ssh_lib => "openssh"}
+      expect(SSHScan::RuleEngine.match({"ssh_lib_eq" => "openssh"}, result)).to be true
+      expect(SSHScan::RuleEngine.match({"ssh_lib_neq" => "dropbear"}, result)).to be true
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.match({"ssh_version_eq" => 1.99}, result)).to be false
+      expect(SSHScan::RuleEngine.match({"ssh_version_neq" => 1.99}, result)).to be true
+    end
+
+    it "gt, lt, gte, lte should work for numeric literal" do
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.match({"ssh_version_gt" => 1.99}, result)).to be true
+      expect(SSHScan::RuleEngine.match({"ssh_version_gte" => 2.0}, result)).to be true
+      expect(SSHScan::RuleEngine.match({"ssh_version_lt" => 1.99}, result)).to be false
+      expect(SSHScan::RuleEngine.match({"ssh_version_lte" => 2.0}, result)).to be true
+    end
+
+    it "should ignore attributes not in results" do
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.match({"ssh_lib_eq" => "openssh"}, result)).to be true
```

Comment:
My expectation here is that you need to have two elements that match before a match would be "true", but the example case is matching true to two different elements.